controllers/base.go

```diff
@@ -174,10 +165,122 @@ func (bmrc *baseMachineReconcileContext) releaseHardware() error {
 	return nil
 }
 
+func (bmrc *baseMachineReconcileContext) getHardwareForMachine(hardware *tinkv1.Hardware) error {
+	namespacedName := types.NamespacedName{
+		Name:      bmrc.tinkerbellMachine.Spec.HardwareName,
+		Namespace: bmrc.tinkerbellMachine.Namespace,
+	}
+
+	if err := bmrc.client.Get(bmrc.ctx, namespacedName, hardware); err != nil {
+		return fmt.Errorf("getting hardware: %w", err)
+	}
+
+	return nil
+}
+
+// createPowerOffJob creates a BMCJob object with the required tasks for hardware power off.
+func (bmrc *baseMachineReconcileContext) createPowerOffJob(hardware *tinkv1.Hardware) error {
+	powerOffAction := rufiov1.HardPowerOff
+	controller := true
+	bmcJob := &rufiov1.BMCJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-poweroff", bmrc.tinkerbellMachine.Name),
```

Comment:
having a consistent name seems good so you don't get two running jobs conflicting, but this is never cleaned up. I don't think you need to do that necessarily in this PR, but can you at least open a GH issue stating that it needs to be thought through?

Reply:
Sure! Will do thanks.